art/attacks/inference/attribute_inference/true_label_baseline.py

```diff
@@ -0,0 +1,171 @@
+# SOFTWARE.
+"""
+This module implements attribute inference attacks.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+from typing import Optional, Union, TYPE_CHECKING
+
+import numpy as np
+from sklearn.neural_network import MLPClassifier
+
+from art.estimators.classification.classifier import ClassifierMixin
+from art.attacks.attack import AttributeInferenceAttack
+from art.utils import check_and_transform_label_format, float_to_categorical, floats_to_one_hot
+
+if TYPE_CHECKING:
+    from art.utils import CLASSIFIER_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class AttributeInferenceBaselineTrueLabel(AttributeInferenceAttack):
+    """
+    Implementation of a baseline attribute inference, not using a model.
+
+    The idea is to train a simple neural network to learn the attacked feature from the rest of the features, and the
+    true label. Should be used to compare with other attribute inference results.
+    """
+
+    _estimator_requirements = ()
+
+    def __init__(
+        self,
+        attack_model: Optional["CLASSIFIER_TYPE"] = None,
+        attack_feature: Union[int, slice] = 0,
+    ):
+        """
+        Create an AttributeInferenceBaseline attack instance.
+
+        :param attack_model: The attack model to train, optional. If none is provided, a default model will be created.
+        :param attack_feature: The index of the feature to be attacked or a slice representing multiple indexes in
+                               case of a one-hot encoded feature.
+        """
+        super().__init__(estimator=None, attack_feature=attack_feature)
+
+        if isinstance(self.attack_feature, int):
+            self.single_index_feature = True
+        else:
+            self.single_index_feature = False
+
+        if attack_model:
+            if ClassifierMixin not in type(attack_model).__mro__:
+                raise ValueError("Attack model must be of type Classifier.")
+            self.attack_model = attack_model
+        else:
+            self.attack_model = MLPClassifier(
+                hidden_layer_sizes=(100,),
+                activation="relu",
+                solver="adam",
+                alpha=0.0001,
+                batch_size="auto",
+                learning_rate="constant",
+                learning_rate_init=0.001,
+                power_t=0.5,
+                max_iter=2000,
+                shuffle=True,
+                random_state=None,
+                tol=0.0001,
+                verbose=False,
+                warm_start=False,
+                momentum=0.9,
+                nesterovs_momentum=True,
+                early_stopping=False,
+                validation_fraction=0.1,
+                beta_1=0.9,
+                beta_2=0.999,
+                epsilon=1e-08,
+                n_iter_no_change=10,
+                max_fun=15000,
+            )
+        # self._scaler = StandardScaler()
```

Comment:
Need to remove these comments. Suggest to use normalizing factor for regression case, same as the black-box attack.